week4/trie/alviprofluvium/Main.hs

```diff
@@ -0,0 +1,29 @@
+module Trie where
+
+import Data.Map.Lazy (Map)
+import qualified Data.Map.Lazy as MapL
+
+data Trie a = Empty | Node (Map a (Trie a))
```

Comment:
As you rightly noticed, this type doesn't let you say whether a certain prefix is a word or not. I changed it in the task to

```
data Trie a = Empty | Node Bool (Map a (Trie a))
```